agreements/notePurchaseAgreementsGateway.go

```diff
@@ -3,54 +3,57 @@ package agreements
 import (
 	"context"
 
+	"github.com/vireocloud/property-pros-service/data"
 	"github.com/vireocloud/property-pros-service/interfaces"
 )
 
 type NotePurchaseAgreementGateway struct {
-	repository interfaces.IAgreementsRepository
+	repository interfaces.IRepository[data.NotePurchaseAgreement]
```

Comment:
This is done to avoid cyclic dependency of interface and data packages.
Currently data package depends on interfaces package, and redefining the IAgreementsResposity interface had dependency on data package for NotePurchaseAgreement model. Ideally the interfaces can be defined in the consumer package to avoid such scenarios.

Reply:
please continue using the interop pattern for cases like this.